pkg/couchdb/couchdb.go

```diff
@@ -251,21 +251,63 @@ func dbNameHasPrefix(dbname, dbprefix string) (bool, string) {
 	return true, strings.Replace(dbname, dbprefix, "", 1)
 }
 
+func buildCouchRequest(db Database, doctype, method, path string, reqjson []byte, headers map[string]string) (*http.Request, error) {
+	if doctype != "" {
+		path = makeDBName(db, doctype) + "/" + path
+	}
+	req, err := http.NewRequest(
+		method,
+		config.CouchURL().String()+path,
+		bytes.NewReader(reqjson),
+	)
+	// Possible err = wrong method, unparsable url
+	if err != nil {
+		return nil, newRequestError(err)
+	}
+	req.Header.Add("Accept", "application/json")
+	if len(reqjson) > 0 {
+		req.Header.Add("Content-Type", "application/json")
+	}
+	for k, v := range headers {
+		req.Header.Add(k, v)
+	}
+	auth := config.GetConfig().CouchDB.Auth
+	if auth != nil {
+		if p, ok := auth.Password(); ok {
+			req.SetBasicAuth(auth.Username(), p)
+		}
+	}
+	return req, nil
+}
+
+func handleResponseError(db Database, resp *http.Response) error {
+	log := logger.WithDomain(db.DomainName()).WithField("nspace", "couchdb")
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		var body []byte
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			err = newIOReadError(err)
+			log.Error(err.Error())
+		} else {
+			err = newCouchdbError(resp.StatusCode, body)
+			log.Debug(err.Error())
+		}
+		return err
+	}
+	return nil
```

Comment:
[nit] I prefer to write this function with a short if (less indentation levels):

```go
func handleResponseError(db Database, resp *http.Response) error {
	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
		return nil
	}

	log := logger.WithDomain(db.DomainName()).WithField("nspace", "couchdb")
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		err = newIOReadError(err)
		log.Error(err.Error())
	} else {
		err = newCouchdbError(resp.StatusCode, body)
		log.Debug(err.Error())
	}
	return err
}
````

Reply:
Right, looks better :+1: